source/slang/ir-insts.h

```diff
@@ -138,12 +123,7 @@ struct IRNameHintDecoration : IRDecoration
 // declaration.
 struct IRDeclRef : IRInst
 {
-    DeclRef<Decl> declRef;
-    virtual void dispose() override
-    {
-        IRInst::dispose();
-        declRef = decltype(declRef)();
-    }
+    Decl* declRef;
```

Comment:
Two notes here:

1. We should confirm whether or not `IRDeclRef` ever gets used. It *shouldn't* be, and a cursory search makes it looks like it is dead code. In that case we should remove it rather than update it.

2. A `DeclRef<Decl>` is *not* equivalent to a `Decl*`, because a `DeclRef` can also hold `Substitutions` needed to specialize a generic `Decl`. This replacement would not be safe/valid if `IRDeclRef` were still in use.